go/consensus/tendermint/apps/staking/messages.go

```diff
@@ -41,6 +42,59 @@ func (app *stakingApplication) changeParameters(ctx *api.Context, msg interface{
 		return nil, fmt.Errorf("staking: failed to validate consensus parameters: %w", err)
 	}
 
+	// Do any necessary state migrations.
+	if changes.MinCommissionRate != nil && apply {
+		var epoch beacon.EpochTime
+		epoch, err = ctx.AppState().GetCurrentEpoch(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("staking: failed to load epoch")
+		}
+		// On MinCommissionRate update, the staking state needs to be updated to ensure all
+		// commission rates and bounds are above the new min commission rate.
+		var addresses []staking.Address
+		addresses, err = state.CommissionScheduleAddresses(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("staking: failed to load addresses: %w", err)
+		}
+		for _, addr := range addresses {
+			var acc *staking.Account
+			acc, err = state.Account(ctx, addr)
+			if err != nil {
+				return nil, fmt.Errorf("staking: failed to load account: %w", err)
+			}
+			var updated bool
+			for i, bound := range acc.Escrow.CommissionSchedule.Bounds {
```

Comment:
What happens if no commission schedule has been set? I guess in this case such an account will not even be returned by `CommissionScheduleAddresses`. We should then probably change the reward distribution (e.g. everything that uses `CurrentRate`) to use the "minimum commission rate" in case the schedule does not exist?

Reply:
Good point. Yeah in case the schedule doesn't exist the minimum rate should be used. 

In that case the migration wouldn't technically be needed, as the `CurrentRate` method (or the code calling it) will need to be aware of the min commission rate anyway, it could just use that in case the configured rate is lower.
The downside is that it could make the configured bounds and rates misleading, so any UIs showing those rates/bounds would need to be aware of the min commission rate. So I think the migration should stay.

Reply:
Yeah I would keep both.

Reply:
Added a `computeCommission` helper method and all the places that compute the commission now use this helper method, which also handles loading the min commission rate parameter in case the rate doesn't exist.